### PR TITLE
new sample using speech API with DLP

### PR DESCRIPTION
**Pull request by blechdom**
also added sallybrown.flac to resources for testing

### CODE REVIEW

**Comment by blechdom** [comment]
> What is the license on the `sallybrown.flac` file?

I recorded it while working at google: unlicensed?

**Comment by gguuss** [comment]
@blechdom Ahh, that should be fine, thank you!

File: speech/cloud-client/transcribe_dlp.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+    config = types.RecognitionConfig(
+        encoding=enums.RecognitionConfig.AudioEncoding.FLAC,
+        sample_rate_hertz=16000,
+        language_code='en-US')
+
+    # Detects speech in the audio file
+    response = speech_client.recognize(config, audio)
+
+    transcript = ""
```
[the remark this answers is not in the document]
Comment (gguuss) [reply speech/cloud-client/transcribe_dlp.py]: Please rewrite this line as:

```python
    transcript = ''
```

File: speech/cloud-client/transcribe_dlp.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+    for result in response.results:
+        transcript = transcript + result.alternatives[0].transcript
```
Comment (gguuss) [inline speech/cloud-client/transcribe_dlp.py]: Prefer format over string concatenation:

```python
transcript = '{}{}'.format(transcript, result.alternatives[0].transcript)

File: speech/cloud-client/transcribe_dlp.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+# Imports the Google Cloud client library
+from google.cloud import speech
+from google.cloud import dlp
+from google.cloud.speech import enums
+from google.cloud.speech import types
+
+
+def deidentify(file_name, project_id):
+    """ convert speech to text and then diedentify entities using dlp """
```
Comment (gguuss) [inline speech/cloud-client/transcribe_dlp.py]: Typo, maybe better written as:

```python
Convert speech to text and then de-identify entities using DLP.
```

File: speech/cloud-client/transcribe_dlp.py [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    # [END dlp_deidentify_masking]
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        description=__doc__,
+        formatter_class=argparse.RawDescriptionHelpFormatter)
+    parser.add_argument(
+        'command',
+        help="deidentify: replace sensitive data with [INFO TYPE]")
```
Comment (gguuss) [inline speech/cloud-client/transcribe_dlp.py]: Single quotes:
```python
help='deidentify : replace sensitive data with [INFO TYPE]'
```

File: speech/cloud-client/transcribe_dlp.py [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+    item = {'value': updated_transcript}
+
+    # Call the API
+    dlp_response = dlp_client.deidentify_content(
+        parent, inspect_config=inspect_config,
+        deidentify_config=deidentify_config, item=item)
+
+    # Print out the results.
+    print('Final Result with sensitive content redacted: {}'
+          .format(dlp_response.item.value))
```
Comment (gguuss) [inline speech/cloud-client/transcribe_dlp.py]: Nit (optional). I prefer to avoid hanging indent, this could also be written as:

```python
print('Final result with sensitive data redacted: {}'
    .format(dlp.response.item.value))
```

**Comment by gguuss** [comment]
@blechdom Looks like you need to sign the cla, please follow the process outlined in the links provided by @googlebot

**Comment by andrewferlitsch** [comment]
@leahecole  Any update on the review?

File: speech/cloud-client/transcribe_dlp.py [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+
+
+def deidentify(file_name, project_id):
+    """ Convert speech to text and then de-identify entities using DLP. """
+    # Instantiates a client
+    speech_client = speech.SpeechClient()
+
+    dlp_client = dlp.DlpServiceClient()
+
+    parent = dlp_client.project_path(project_id)
```
Comment (kurtisvg) [inline speech/cloud-client/transcribe_dlp.py]: What is `parent`? Can we provide some additional context on what this means?

File: speech/cloud-client/transcribe_dlp.py [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+
+    # Prepare info_types by converting the list of strings into a list of
```
Comment (kurtisvg) [inline speech/cloud-client/transcribe_dlp.py]: list of what?

File: speech/cloud-client/transcribe_dlp.py [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+    info_types = ['ALL_BASIC']
+    # dictionaries (protos are also accepted).
```
Comment (kurtisvg) [inline speech/cloud-client/transcribe_dlp.py]: This comment isn't helpful - what does this mean?

File: speech/cloud-client/transcribe_dlp.py [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+        'info_types': [{'name': info_type} for info_type in info_types]
+    }
+    # Construct deidentify configuration dictionary
+    deidentify_config = {
+        'info_type_transformations': {
+            'transformations': [
+                {
+                    'primitive_transformation': {
+                        'replace_with_info_type_config': {
+
```
Comment (kurtisvg) [inline speech/cloud-client/transcribe_dlp.py]: Why is this empty?

File: speech/cloud-client/transcribe_dlp.py [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+    parent = dlp_client.project_path(project_id)
+
+    # Prepare info_types by converting the list of strings into a list of
+    info_types = ['ALL_BASIC']
+    # dictionaries (protos are also accepted).
+    inspect_config = {
+        'info_types': [{'name': info_type} for info_type in info_types]
+    }
+    # Construct deidentify configuration dictionary
```
Comment (kurtisvg) [inline speech/cloud-client/transcribe_dlp.py]: This comment is also unhelpful - it is a literal redefinition of the following code. Can we update to explain why or what the user is doing exactly?

File: speech/cloud-client/transcribe_dlp.py [hunk 11]
```diff
@@ -0,0 +1,140 @@
...
+    # Format with regex before sending to DLP api
+    # Currently social security numbers and credit card numbers
+    # are interpreted as phone numbers
+
+    regex = (
+             r".([A-Za-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[A-Za-z0-9!#$%&'*+/=?^_`"
+             r"{|}~-]+)*)(\sat\s+)((?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+"
+             r"[a-z0-9](?:[a-z0-9-]*[a-z0-9]))"
+    )
+
```
Comment (kurtisvg) [inline speech/cloud-client/transcribe_dlp.py]: nit: There is a lot of unnecessary whitespace between lines that makes this harder to read. Can we try to group related commands a bit more closely?

**Comment by andrewferlitsch** [comment]
@beccasaurus Kurtis is looking for someone to write tests for these samples.

**Comment by kurtisvg** [comment]
Closing this PR as it has gone stale.
